pkg/deploy/controller/imagechange/controller.go

```diff
@@ -3,161 +3,124 @@ package imagechange
 import (
 	"fmt"
 
+	"k8s.io/kubernetes/pkg/util/sets"
+
 	"github.com/golang/glog"
 
+	"github.com/openshift/origin/pkg/client"
 	deployapi "github.com/openshift/origin/pkg/deploy/api"
 	deployutil "github.com/openshift/origin/pkg/deploy/util"
 	imageapi "github.com/openshift/origin/pkg/image/api"
 )
 
-// ImageChangeController increments the version of a DeploymentConfig which has an image
+// ImageChangeController increments the version of a deployment config which has an image
 // change trigger when a tag update to a triggered ImageStream is detected.
 //
 // Use the ImageChangeControllerFactory to create this controller.
 type ImageChangeController struct {
-	deploymentConfigClient deploymentConfigClient
+	listDeploymentConfigs func() ([]*deployapi.DeploymentConfig, error)
+	client                client.Interface
 }
 
 // fatalError is an error which can't be retried.
 type fatalError string
 
 func (e fatalError) Error() string {
-	return fmt.Sprintf("fatal error handling ImageStream: %s", string(e))
+	return fmt.Sprintf("fatal error handling image stream: %s", string(e))
 }
 
-// Handle processes image change triggers associated with imageRepo.
-func (c *ImageChangeController) Handle(imageRepo *imageapi.ImageStream) error {
-	configs, err := c.deploymentConfigClient.listDeploymentConfigs()
+// Handle processes image change triggers associated with imagestream.
+func (c *ImageChangeController) Handle(stream *imageapi.ImageStream) error {
+	configs, err := c.listDeploymentConfigs()
 	if err != nil {
-		return fmt.Errorf("couldn't get list of DeploymentConfig while handling ImageStream %s: %v", labelForRepo(imageRepo), err)
+		return fmt.Errorf("couldn't get list of deployment configs while handling image stream %q: %v", imageapi.LabelForStream(stream), err)
 	}
 
 	// Find any configs which should be updated based on the new image state
-	configsToUpdate := map[string]*deployapi.DeploymentConfig{}
+	configsToUpdate := []*deployapi.DeploymentConfig{}
 	for _, config := range configs {
-		glog.V(4).Infof("Detecting changed images for DeploymentConfig %s", deployutil.LabelForDeploymentConfig(config))
+		glog.V(4).Infof("Detecting image changes for deployment config %q", deployutil.LabelForDeploymentConfig(config))
+		hasImageChange := false
 
 		for _, trigger := range config.Spec.Triggers {
 			params := trigger.ImageChangeParams
 
 			// Only automatic image change triggers should fire
-			if trigger.Type != deployapi.DeploymentTriggerOnImageChange || !params.Automatic {
+			if trigger.Type != deployapi.DeploymentTriggerOnImageChange {
 				continue
 			}
 
-			// Check if the image repo matches the trigger
-			if !triggerMatchesImage(config, params, imageRepo) {
+			// Resolve images for all initial deployments.
```

Comment:
It's not inaccurate - maybe ambiguous? All initial deployments will have their images resolved, even those with automatic==false (fixes https://github.com/openshift/origin/issues/6934) . Otherwise (latestVersion != 0), we are going to resolve only automatic==true triggers. I will update the comment clarifying this.